pkg/ccl/backupccl/backup_job.go

```diff
@@ -664,18 +681,229 @@ func (b *backupResumer) readManifestOnResume(
+	if err != nil {
+		return errors.Wrap(err, "getting spans to protect")
+	}
+
+	// Protect the spans after the EndTime of the current backup. We do not need
+	// to verify this new record as we have a record written by the backup during
+	// planning, already protecting these spans after EndTime.
+	//
+	// Since this record will be stored on the incremental schedule, we use the
+	// inc schedule ID as the records' Meta. This ensures that even if the full
+	// schedule is dropped, the reconciliation job will not release the pts
+	// record stored on the inc schedule, and the chaining will continue.
+	ptsRecord, err := protectTimestampRecordForSchedule(ctx, spansToProtect,
+		backupDetails.EndTime, incSj.ScheduleID(), exec, txn)
+	if err != nil {
+		return errors.Wrap(err, "protect and verify pts record for schedule")
+	}
+
+	// Attempt to release the pts record that was written on the incremental
+	// schedule when the full backup was being planned.
+	if err := releaseProtectedTimestampRecordForSchedule(ctx, txn,
+		backupDetails.ReleaseOnSuccessPTSRecord, exec); err != nil {
+		return errors.Wrap(err, "release pts record for schedule")
+	}
+
+	// Update the incremental schedule with the new pts record.
+	incSj.ExecutionArgs().ProtectedTimestampRecord = &ptsRecord
+	incSj.SetExecutionDetails(
+		incSj.ExecutorType(),
+		*incSj.ExecutionArgs(),
+	)
+	return incSj.Update(ctx, exec.InternalExecutor, txn)
+}
+
+func (b *backupResumer) maybeUpdateProtectedTimestampRecordOnSchedule(
+	ctx context.Context, exec *sql.ExecutorConfig,
+) error {
+	env := scheduledjobs.ProdJobSchedulerEnv
+	if knobs, ok := exec.DistSQLSrv.TestingKnobs.JobsTestingKnobs.(*jobs.TestingKnobs); ok {
+		if knobs.JobSchedulerEnv != nil {
+			env = knobs.JobSchedulerEnv
+		}
+	}
+
+	var backupDetails jobspb.BackupDetails
+	var ok bool
+	if backupDetails, ok = b.job.Details().(jobspb.BackupDetails); !ok {
+		return errors.Newf("unexpected job details type %T", b.job.Details())
+	}
+
+	if err := exec.DB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+		// We cannot rely on b.job containing created_by_id because on job
+		// resumption the registry does not populate the resumers' CreatedByInfo.
+		datums, err := exec.InternalExecutor.QueryRowEx(
+			ctx,
+			"lookup-schedule-info",
+			txn,
+			sessiondata.InternalExecutorOverride{User: security.NodeUserName()},
+			fmt.Sprintf(
+				"SELECT created_by_id FROM %s WHERE id=$1 AND created_by_type=$2",
+				env.SystemJobsTableName()),
+			b.job.ID(), jobs.CreatedByScheduledJobs)
+
+		if err != nil {
+			return errors.Wrap(err, "schedule info lookup")
+		}
+		if datums == nil {
+			// Not a scheduled backup.
+			return nil
+		}
+
+		scheduleID := int64(tree.MustBeDInt(datums[0]))
+		_, args, err := getScheduledBackupExecutionArgsFromSchedule(ctx, env, txn,
+			exec.InternalExecutor, scheduleID)
+		if err != nil {
+			return errors.Wrap(err, "load scheduled job")
```

Comment:
Do we want to error if the schedule isn't found? What if the schedule was dropped during execution?

Reply:
Hmm if the schedule is dropped during backup execution then `NotifyJobTermination` at the end of `Resume` will also probably return an error and fail the job. I made OnFailOrCancel not fail in such a scenario but maybe it is okay to fail a backup if you decide to nuke the schedule that owns it?

Reply:
fyi: We could extend drop schedule to disallow drops when active jobs sitll running.

Reply:
Filed https://github.com/cockroachdb/cockroach/issues/69129 to track some of these DROP SCHEDULE changes. We can clean this code up once some of those items are addressed.

Reply:
Thanks for filing. I think the main thing here is to have a consistent story around dropping schedules. In my head, schedules were the things that created jobs, so if I drop a schedule I would expect future jobs to no longer be created, so NotifyTermination callbacks may noop but I would intuitively be a bit surprised if the other jobs failed.

Like, if the schedule isn't found I think it's safe to assume that it is not coming back so it seems safe to early return?